src/pages/settings/addressbook/add/add.ts

```diff
@@ -82,9 +82,31 @@ export class AddressbookAddPage {
   }
 
   public save(): void {
-    this.addressBookAdd.controls['address'].setValue(
-      this.parseAddress(this.addressBookAdd.value.address)
+    let newAddress: string = this.parseAddress(
+      this.addressBookAdd.value.address
     );
+
+    this.addressBookAdd.controls['address'].setValue(newAddress);
+    let newAddressOrder: number;
+
+    let promises = [];
+    promises.push(
```

Comment:
pushing the promises is not needed here

Reply:
This has been removed.

> Nice work ! could you make it work for the send view so that the order is respected in the contacts card?

The contacts card now shows the respected order for the address book in the next commit.